examples/nonlinear_eady.py

```diff
@@ -0,0 +1,168 @@
+
+# vertical coordinate and normal
+x = SpatialCoordinate(mesh)
+z = Function(W_CG1).interpolate(x[2])
+k = Function(W_VectorCG1).interpolate(Expression(("0.","0.","1.")))
+
+# Coriolis expression
+f = 1.e-04
+Omega = as_vector([0.,0.,f*0.5])
+
+# list of prognostic fieldnames
+# this is passed to state and used to construct a dictionary,
+# state.field_dict so that we can access fields by name
+# u is the 3D velocity
+# p is the pressure
+# b is the buoyancy
+fieldlist = ['u', 'p', 'b']
+
+# class containing timestepping parameters
+# all values not explicitly set here use the default values provided
+# and documented in configuration.py
+timestepping = TimesteppingParameters(dt=dt)
+
+# class containing output parameters
+# all values not explicitly set here use the default values provided
+# and documented in configuration.py
+output = OutputParameters(dirname='nonlinear_eady', dumpfreq=12*36, dumplist=['u','p'])
+
+# class containing physical parameters
+# all values not explicitly set here use the default values provided
+# and documented in configuration.py
+parameters = EadyParameters(H=H)
+
+# class for diagnostics
+# fields passed to this class will have basic diagnostics computed
+# (eg min, max, l2 norm) and these will be output as a json file
+diagnostics = Diagnostics(*fieldlist)
+
+# list of diagnostic fields, each defined in a class in diagnostics.py
+diagnostic_fields = [CourantNumber()]
+
+# setup state, passing in the mesh, information on the required finite element
+# function spaces, z, k, and the classes above
+state = IncompressibleState(mesh, vertical_degree=1, horizontal_degree=1,
+                            family="RTCF",
+                            z=z, k=k, Omega=Omega,
+                            timestepping=timestepping,
+                            output=output,
+                            parameters=parameters,
+                            diagnostics=diagnostics,
+                            fieldlist=fieldlist,
+                            diagnostic_fields=diagnostic_fields,
+                            on_sphere=False)
+
+##############################################################################
+# Initial conditions
+##############################################################################
+# set up functions on the spaces constructed by state
+u0, p0, b0 = Function(state.V[0]), Function(state.V[1]), Function(state.V[2])
+
+# first setup the background buoyancy profile
+# z.grad(bref) = N**2
+# the following is symbolic algebra, using the default buoyancy frequency
+# from the parameters class. x[2]=z and comes from x=SpatialCoordinate(mesh)
+Nsq = parameters.Nsq
+bref = x[2]*Nsq
+# interpolate the expression to the function
+b_b = Function(state.V[2]).interpolate(bref)
+
+# setup constants
+a = -7.5
+Bu = 0.5
+template_s = inifns.template_target_strings()
+b_exp = Expression(template_s,a=a,Nsq=Nsq,Bu=Bu,H=H,L=L)
+b_pert = Function(state.V[2]).interpolate(b_exp)
```

Comment:
You can [interpolate UFL expressions](http://firedrakeproject.org/interpolation.html#ufl-expressions), so I think there's no need for this sympy stuff.   Something like:

```python
def coth(x):
    return cosh(x)/sinh(x)
x, y, z = SpatialCoordinate(mesh)
b_pert = Function(state.V[2]).interpolate(tanh(x) * coth(x))
```

for example (I think your sympy code just just translate directly.